### PR TITLE
Added test for using numba.jit with lambdify

### PR DESCRIPTION
**Pull request by xavierdurawa**
Added a test to check that numba.jit is working
properly with lambdified functions.

#### References to other Issues or PRs

Addresses #20070

#### Brief description of what is fixed or changed
New test was added to ensure that numba.jit works
with Sympy's lambdified functions

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/external/tests/test_numba.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    f2 = lambdify(x, ddx_tanh, 'numpy')
+    assert numba.jit(f2)(1) == f2(1)
+
+    from itertools import chain
+    X = Matrix(symbols('x:3')).T
+    Y = Matrix(symbols('y:3')).T
+    f3 = lambdify((*X, *Y), X+Y)
+    xs = (1,2,3,4,5,6)
+    assert all(chain.from_iterable(f3(*xs) == numba.njit(f3)(*xs)))
+
```
Comment (eric-wieser) [inline sympy/external/tests/test_numba.py:29]:
```suggestion
```

Comment (xavierdurawa) [reply sympy/external/tests/test_numba.py:29]: As in add numba to .travis.yaml?

Comment (eric-wieser) [reply sympy/external/tests/test_numba.py:29]: No, removing the blank line

Comment (asmeurer) [reply sympy/external/tests/test_numba.py:29]: You also should add it to Travis.

File: sympy/external/tests/test_numba.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+numba = import_module('numba')
+if not numba:
+    #bin/test will not execute any tests now
+    disabled = True
+
+def test_jit():
+    # Test that numba.jit works with lambdify
+    x = symbols('x')
+    f1 = lambdify(x, sin(x), 'numpy')
+    assert numba.jit(f1)(1) == f1(1)
```
Comment (asmeurer) [inline sympy/external/tests/test_numba.py:15]: Numba jitted functions don't get the exact same numeric result as NumPy in general. You may want to use `assert abs(numba.jit(f1)(1) - f1(1)) < 1e-15` instead.

File: sympy/external/tests/test_numba.py (hunk 1 above)
Comment (xavierdurawa) [reply sympy/external/tests/test_numba.py]: not sure if I added numba to Travis correctly.  If I didn't, a suggestion or point in the right direction for how to do that would be appreciated.  Thanks

Comment (asmeurer) [reply sympy/external/tests/test_numba.py]: You did, but it looks like you also need to add 

```
# numba
'*numba*',
```

here https://github.com/sympy/sympy/blob/master/bin/test_travis.sh#L188. You can check in the Travis build, this file should be one of the ones tested in the TEST_OPT_DEPENDENCY builds.

File: sympy/external/tests/test_numba.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    ddx_tanh = tanh.diff(x)
+    f2 = lambdify(x, ddx_tanh, 'numpy')
+    assert abs(numba.jit(f2)(1) - f2(1)) < 1e-15
+
+    from itertools import chain
+    X = Matrix(symbols('x:3')).T
+    Y = Matrix(symbols('y:3')).T
+    f3 = lambdify((*X, *Y), X+Y)
+    xs = (1,2,3,4,5,6)
+    element_comp = abs(f3(*xs) - numba.njit(f3)(*xs) < 1e-15)
```
Comment (oscarbenjamin) [inline sympy/external/tests/test_numba.py:28]: This is calling `abs` on the bool array. The `1e-15` shouldn't be inside the bracket.

**Comment by czgdp1807** [comment]
I have restarted the tests. @oscarbenjamin @asmeurer Are there any more changes needed to be made for this PR to be merged?

**Pull request closed without merging**